naas/requests/project_subscriber_profiles.py

```diff
@@ -0,0 +1,67 @@
+import json
+from naas.client import Client
+
+
+class ProjectSubscriberProfiles:
+
+    @staticmethod
+    def retrieve_by_project_id_and_project_subscriber_id(
+            project_id, project_subscriber_id, params=None):
+        """
+        Retrieve the instance of a project subscriber by project
+
+        :param project_id: int
+        :param project_subscriber_id: int
+        :param params: dict
+        :return: Response
+        """
+        if params is None:
+            params = {}
+
+        rel = Client.rel_for('rels/project-property')
+        route = Client.routes().route_for(rel)
+        url = route.url_for(
+            args={
+                **params,
+                **{
+                    'project_id': project_id,
+                    'project_subscriber_id': project_subscriber_id
+                }
+            }
+        )
+        request = Client.get(url)
+        return request
+
+    @staticmethod
+    def update_by_project_id_and_project_subscriber_id(
+            project_id, project_subscriber_id, params=None):
+        """
+        Update an existing record
+
+        :param project_id: int
+        :param project_subscriber_id: int
+        :param params: dict
+        :return: Response
+        """
+        if params is None:
+            params = {}
```

Comment:
@arthuroe - Curious about this pattern as I don't write as much python as I do ruby. Is there a specific reason we wouldn't set this as a default in the method definition:

```
def update_by_project_id_and_project_subscriber_id(project_id, project_subscriber_id, params={})
```

We would never want this to be `None`, and we know we want to work with a hash. It seems like it would eliminate the need to check and see if it is `None` as it would be the default hash.

Reply:
Yeah in python we try to avoid defining mutable objects in arguments. We do the check to ensure we do not pass a None item.

Reply:
@arthuroe - thank you! I learned something new. I had assumed that new instances of objects would be fresh starts. This is quite the gotcha (and seems very strange).